### PR TITLE
Implement core downcount hack

### PR DESCRIPTION
**Pull request by Gamer64ytb**
When enabled, core downcount will be limited to a smaller time slice, reducing CPU usage. May cause performance issues for some games, or may improve for another games. Benefits a lot for some games as luigi's mansion 2 if you also want to upscale it.

Some of this hack base is from Citra MMJ and its hack "CPU Usage Limit", rest is reworked from me.

### CODE REVIEW

**Comment by BlurrySquire** [comment]
What does the core downcount do? I'm not sure what it does exactly but I'm interested.

**Comment by OpenSauce04** [comment]
Ditto on that; I tried to look up "core downcount" to discover what you're talking about, and this pull request showed up in the results instead
An explanation would be nice

**Comment by rtiangha** [comment]
Considering this came from Citra MMJ (or Citra on Android? I'm not sure what its name is), the hint on what this does is in the code comments. The descriptions are spread out, so it's worth skimming through the whole file to get a handle on what this does and whether it's worth including here.

https://github.com/weihuoya/citra/blob/b99cb4f3c92311388b8eb659a91347c668502564/src/core/core.cpp#L49

Edit: Oh, looks like the downcount comments are already in Master. My bad. Anyway, my understanding is by default, Master does this:

> Now all cores are at the same global time. So we will run them one after the other with a max slice that is the minimum of all max slices of all cores

And what the hack does is reduce that time slice even more. So as it says, "When enabled, core downcount will be limited to a smaller time slice, reducing CPU usage. May cause performance issues for some games, or may improve for another games"

I find that I sometimes get a few fps extra in some games, but if my fps is too slow to begin with (ex. 45 out of 60 fps or 23 out of 30 fps), I lose audio if things aren't vsynced for too long (as in it literally dies after a while). And some games don't start with it enabled. So it's one of those "try it and see if it helps or hurts" options.

**Comment by OpenSauce04** [comment]
@Gamer64ytb Please update your commit history to properly attribute the original author. I am blocking this PR until this has been done

**Comment by rtiangha** [comment]
FYI, it'd be easier to do that on the command line. So something like
`git commit --author="AUTHOR NAME <emailaddress@gmail.com>" -m "INSERT COMMIT MESSAGE"`

Edit: changed message to remove personal info (and it wasn't the relevant author anyway so don't copy/paste).

**Comment by DonelBueno** [comment]
Tested on Luigi's Mansion Dark Moon and all it does to "speed up" the game is to reduce the emulated framerate from 60 to 30-40.

**Comment by OpenSauce04** [comment]
The overall consensus on the performance of tests seems to indicate that this may not be a positive change even without the attribution issue

**Comment by OpenSauce04** [comment]
The overall consensus seems to indicate that this may not be a positive change even without the attribution issue
@Polarzincomfrio @BlurrySquire what are your thoughts on this?

**Comment by Gamer64ytb** [comment]
As I said, its useful for weak devices mainly.

**Comment by BlurrySquire** [comment]
> The overall consensus seems to indicate that this may not be a positive change even without the attribution issue @Polarzincomfrio @BlurrySquire what are your thoughts on this?

From my testing it turns a solid 60fps in Luigi's Mansion: Dark Moon into a 30-40 fps experience from my testing. It seems more like an option for reducing CPU load, which I don't think it does either. It just makes the game run slower.

Most of the performance issues for Luigi's Mansion: Dark Moon and others have been fixed by porting commits from PabloMK7/Citra (with co-authors). The only way to make it work better on lower end devices without losing quality (lower framerate/speed) is to optimise.

Regarding the attribution issue, it hasn't even been fixed yet, even though @rtiangha explained a way to fix the issue 2 hours ago. It was a simple fix as well, a single git command. I understand people can be busy but that fix would take no more than 10 minutes.

In conclusion, I think we scrap this pull request. From developer testing, there has only been a drop in the framerate with the option enabled. We don't want to violate GPLv2 so we can't merge this pull request, when we know the code is taken from somewhere else without attribution.

**Comment by CheriCheriLady87** [comment]
> As I said, its useful for weak devices mainly.

Please show us proof that this actually happens on low-end devices,  I do not know if Galaxy A34 is low-end enough for you, but i didn't notice anytback when i tested this on the other fork

**Comment by OpenSauce04** [comment]
I will be closing this PR due to a combination of issues such as the lack of real-world results, the hacky nature of the changes, and the attribution issue
